backend/src/test/java/shook/shook/song/application/SongSearchServiceTest.java

```diff
@@ -0,0 +1,182 @@
+package shook.shook.song.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import shook.shook.song.application.dto.SearchedSongResponse;
+import shook.shook.song.domain.Song;
+import shook.shook.song.domain.repository.SongRepository;
+import shook.shook.support.UsingJpaTest;
+
+class SongSearchServiceTest extends UsingJpaTest {
+
+    @Autowired
+    private SongRepository songRepository;
+
+    private SongSearchService songSearchService;
+
+    @BeforeEach
+    void setUp() {
+        songSearchService = new SongSearchService(songRepository);
+    }
+
+    @DisplayName("제목이 비어있을 때")
+    @Nested
+    class SongSingerSearch {
+
+        @DisplayName("앞뒤 공백제거, 대소문자 상관없이 정확히 일치하는 가수 이름의 모든 노래를 조회한다.")
+        @ParameterizedTest(name = "가수의  이름이 {0} 일 때")
+        @ValueSource(strings = {"   redvelvet ", "	Redvelvet\n"})
+        void findAllBySinger_exist(final String singer) {
+            //given
+            final Song saved2 = songRepository.save(new Song("아무노래", "비디오URL", "RedVelvet", 180));
+            songRepository.save(new Song("다른노래", "비디오URL", "Red Velvet", 180));
+
+            //when
+            saveAndClearEntityManager();
+            final List<SearchedSongResponse> responses = songSearchService.findAllBySingerAndTitle(
+                singer, null);
+
+            //then
+            final List<SearchedSongResponse> expectedResponses = Stream.of(saved2)
+                .map(SearchedSongResponse::from)
+                .toList();
+
+            assertThat(responses).usingRecursiveComparison()
+                .isEqualTo(expectedResponses);
+        }
+
+        @DisplayName("정확히 일치하는 가수 이름 조회 시, 일치 결과가 없다면 빈 내용이 반환된다.")
+        @Test
+        void findAllBySinger_noExist() {
+            //given
+            //when
+            final List<SearchedSongResponse> responses = songSearchService.findAllBySingerAndTitle(
+                "가수2", null);
+
+            //then
+            assertThat(responses).usingRecursiveComparison()
+                .isEqualTo(Collections.emptyList());
```

Comment:
💬 혹시 isEmpty 는 어떠신가요~

```suggestion
            assertThat(responses).isEmpty();
```

Reply:
너무 좋다고 생각합니다 👍 👍